### PR TITLE
Python packages set: patch updates

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change

```
$ maintainers/scripts/update-python-libraries --use-pkgs-prefix --commit --target=patch pkgs/development/python-modules/**/*.nix
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/boto3/default.nix [hunk 1]
```diff
@@ -13,11 +13,11 @@
 
 buildPythonPackage rec {
   pname =  "boto3";
-  version = "1.15.9"; # N.B: if you change this, change botocore too
+  version = "1.15.18"; # N.B: if you change this, change botocore too
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/boto3/default.nix:16]:
```suggestion
  version = "1.15.18"; # N.B: if you change this, change boto3 and awscli to a matching version
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Did you use a script to do this? Can I find it somewhere?

**Comment by FRidh** [comment]
@SuperSandro2000 
```
$ maintainers/scripts/update-python-libraries --use-pkgs-prefix --commit --target=patch pkgs/development/python-modules/**/*.nix
```

**Comment by mweinelt** [comment]
The zipgy updates collide with #101557 and I'd like the maintainer to work on test integration in that pull request. They also require an update to home-assistants `component-packages.nix`, so I'd appreciate if you could drop them here.

**Comment by mweinelt** [comment]
```
builder for '/nix/store/zg8jl91486n45sz196s2z3zmxs74g36m-python3.8-amply-0.1.4.drv' failed with exit code 1; last 10 log lines:
  patching script interpreter paths in /nix/store/c81cvqg9yjzhvf3ff5b25kgxwzc0g8l2-python3.8-amply-0.1.4
  checking for references to /build/ in /nix/store/c81cvqg9yjzhvf3ff5b25kgxwzc0g8l2-python3.8-amply-0.1.4...
  Executing pythonRemoveTestsDir
  Finished executing pythonRemoveTestsDir
  running install tests
  Traceback (most recent call last):
    File "tests/test_amply.py", line 4, in <module>
      import pytest
  ModuleNotFoundError: No module named 'pytest'
  builder for '/nix/store/zg8jl91486n45sz196s2z3zmxs74g36m-python3.8-amply-0.1.4.drv' failed with exit code 1
```

https://github.com/willu47/amply/blame/v0.1.4/tests/test_amply.py#L4

**Comment by jonringer** [comment]
fridh, your pip reproducible patch isn't included in here. many packages are failing

**Comment by jonringer** [comment]
@fridh I rebased this on top of the latest to bringing your fixes for builds affected by the pip reproducibility patch.

Unfortunately, I'm still affected by https://github.com/xianyi/OpenBLAS/issues/2970 and unable to build numpy. So, I'll need to wait for hydra to build it for me to progress much further

**Comment by jonringer** [comment]
this is probably more stable than master at this point

**Comment by raboof** [comment]
The pytest-testmon 1.0.2 -> 1.0.3 included in this change appears to fail consistently.

I'm no python programmer, but indeed looking at the [PyPI package](https://pypi.org/project/pytest-testmon/#files), `test/test_core.py:131` expects `filenames_fingerprints` to contain 5 elements, but the query at `testmon/testmon_core.py:267` only selects 4.

I don't see any tests in the source repo at https://github.com/tarpas/pytest-testmon, though, so it's unclear to me where these tests actually come from...

**Comment by raboof** [comment]
> The pytest-testmon 1.0.2 -> 1.0.3 included in this change appears to fail consistently.

It seems the tests are expecting master instead of 1.0.3 (https://github.com/tarpas/pytest-testmon/issues/158), let's skip them for now (https://github.com/NixOS/nixpkgs/pull/103786)

**Pull request merged**